var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -185,8 +185,8 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('petsc+double~complex', when='+petsc precision=double')
 
     # Require that external libraries built with the same index type
-    depends_on('hypre~int64', when='+hypre ~int64')
-    depends_on('hypre+int64', when='+hypre +int64')
+    depends_on('hypre@:2.22.0~int64', when='@:5.7.0 +hypre ~int64')
```

Comment:
I think these two lines can stay as they were. The `depends_on` above should handle the version.

Reply:
It does not work for me with the original concretizer
`spack install sundials+hypre+int64`

> ==> Error: An unsatisfiable version constraint has been detected for spec:
> sundials requires hypre version :2.22.0, but spec asked for 2.22.1

Reply:
What about
```
    depends_on('hypre~int64', when='@5.7.1: +hypre ~int64')
    depends_on('hypre+int64', when='@5.7.1: +hypre +int64')
    depends_on('hypre@:2.22.0~int64', when='@:5.7.0 +hypre ~int64')
    depends_on('hypre@:2.22.0+int64', when='@:5.7.0 +hypre +int64')
```

Reply:
That is close, but we want to have `hypre+mpi` too. So I think the complete diff (with whitespace changes) needed is:

```
diff --git a/var/spack/repos/builtin/packages/sundials/package.py diff --git a/var/spack/repos/builtin/packages/sundials/package.py b/var/spack/repos/builtin/packages/sundials/package.py
index cd65d6ac8c..fda3611225 100644
--- a/var/spack/repos/builtin/packages/sundials/package.py
+++ b/var/spack/repos/builtin/packages/sundials/package.py
@@ -172,21 +172,22 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('raja+rocm', when='+raja +rocm')
 
     # External libraries
-    depends_on('lapack',              when='+lapack')
-    depends_on('suite-sparse',        when='+klu')
-    depends_on('petsc+mpi',           when='+petsc')
-    depends_on('hypre+mpi',           when='+hypre')
-    depends_on('superlu-dist@6.1.1:', when='@:5.4.0 +superlu-dist')
-    depends_on('superlu-dist@6.3.0:', when='@5.5.0: +superlu-dist')
-    depends_on('trilinos+tpetra',     when='+trilinos')
+    depends_on('lapack',                  when='+lapack')
+    depends_on('suite-sparse',            when='+klu')
+    depends_on('petsc+mpi',               when='+petsc')
+    depends_on('hypre+mpi~int64',         when='@5.7.1: +hypre ~int64')
+    depends_on('hypre+mpi+int64',         when='@5.7.1: +hypre +int64')
+    depends_on('hypre@:2.22.0+mpi~int64', when='@:5.7.0 +hypre ~int64')
+    depends_on('hypre@:2.22.0+mpi+int64', when='@:5.7.0 +hypre +int64')
+    depends_on('superlu-dist@6.1.1:',     when='@:5.4.0 +superlu-dist')
+    depends_on('superlu-dist@6.3.0:',     when='@5.5.0: +superlu-dist')
+    depends_on('trilinos+tpetra',         when='+trilinos')
 
     # Require that external libraries built with the same precision
     depends_on('petsc~double~complex', when='+petsc precision=single')
     depends_on('petsc+double~complex', when='+petsc precision=double')
 
     # Require that external libraries built with the same index type
-    depends_on('hypre~int64', when='+hypre ~int64')
-    depends_on('hypre+int64', when='+hypre +int64')
     depends_on('petsc~int64', when='+petsc ~int64')
     depends_on('petsc+int64', when='+petsc +int64')
     depends_on('superlu-dist+int64', when='+superlu-dist +int64')
```

Reply:
That is right, it is simpler. So I made that change. Thanks.